### PR TITLE
doc: Updated 54H memory arch doc

### PR DESCRIPTION
**Pull request by FrancescoSer**
Updated 54H memory arch doc.

### CODE REVIEW

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_architecture_memory.rst [hunk 1]
```diff
@@ -42,19 +42,20 @@ Local RAM is present in each of local domains
 Application core RAM
 --------------------
 
-.. image:: images/nrf54h20_memory_map_app.svg
-   :width: 300 px
+.. table:: Application Core RAM layout in the nRF54H20
```
Comment (annwoj) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_architecture_memory.rst:45]:
```suggestion
.. table:: Application core RAM layout in the nRF54H20
```
To align with other occurences

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_architecture_memory.rst [hunk 2]
```diff
@@ -70,17 +71,20 @@ Access control
 Radio core RAM
 --------------
 
-The radio core contains 96 KB of local RAM.
+.. table:: Radio Core RAM layout in the nRF54H20
```
Comment (annwoj) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_architecture_memory.rst:74]:
```suggestion
.. table:: Radio core RAM layout in the nRF54H20
```

**Review by annwoj — APPROVED** [review]

File: doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_architecture_memory.rst [hunk 3]
```diff
@@ -42,19 +42,20 @@ Local RAM is present in each of local domains
...
 --------------------
 
-.. image:: images/nrf54h20_memory_map_app.svg
-   :width: 300 px
+.. table:: Application core RAM layout in the nRF54H20
+
+   +-------------+-----------+---------+----------+-----------------------------+
+   |Start Address|End Address|Size (KB)|Total (KB)|Content                      |
+   +=============+===========+=========+==========+=============================+
+   |config       |0x22008000 |config   |32        |Application's Non-Secure data|
```
Comment (hubertmis) [inline doc/nrf/app_dev/device_guides/working_with_nrf/nrf54h/ug_nrf54h20_architecture_memory.rst:50]: I think we should define in this document what "config" means. (it means configurable by an application developer to meet the application needs).

**Review by hubertmis — APPROVED** [review]

**Comment by hubertmis** [comment]
@maciejpietras, FYI

**Pull request merged**